src/js/components/deployments/inprogressdeployments.js

```diff
@@ -70,18 +70,14 @@ export class Progress extends React.Component {
   }
 
   // // deploymentStatus = <inprogress|pending>
-  refreshDeployments(page, perPage, deploymentStatus, fullRefresh = true) {
+  refreshDeployments(page, perPage, deploymentStatus) {
     const self = this;
     const mappedDeploymentStatus = deploymentStatusMap[deploymentStatus];
-    return self.setState({ [`${mappedDeploymentStatus}Page`]: page, [`${mappedDeploymentStatus}PerPage`]: perPage }, () => {
-      let tasks = [self.props.getDeploymentCount(deploymentStatus)];
-      if (fullRefresh) {
-        tasks.push(self.props.getDeploymentsByStatus(deploymentStatus, page, perPage));
-      }
-      return Promise.all(tasks)
-        .then(([countAction, deploymentsAction]) => {
+    return self.setState({ [`${mappedDeploymentStatus}Page`]: page, [`${mappedDeploymentStatus}PerPage`]: perPage }, () =>
+      Promise.resolve(self.props.getDeploymentsByStatus(deploymentStatus, page, perPage))
+        .then(deploymentsAction => {
           clearRetryTimer(deploymentStatus, self.props.setSnackbar);
-          if (countAction.deploymentIds.length && deploymentsAction && !deploymentsAction[0].deploymentIds.length) {
+          if (deploymentsAction && deploymentsAction[0].total && !deploymentsAction[0].deploymentIds.length) {
```

Comment:
I would check length here

Reply:
I would choose to disregard this, since the actions either return a single object or an array if they are a) returning a promise and b) successful - otherwise the returned value will be undefined... until now the returned object depends on the action called (so one of <promise returning object|promise returning array|plain object of dispatched action>) so I'm confident enough that this is aligned as long as the UI doesn't blow up the first time it's called 🤷‍♂️